internal/executor/instance/persistentworker/persistentworker.go

```diff
@@ -37,11 +56,28 @@ func New(isolation *api.Isolation, logger logger.Lightweight) (abstract.Instance
 		return parallels.New(iso.Parallels.Image, iso.Parallels.User, iso.Parallels.Password,
 			strings.ToLower(iso.Parallels.Platform.String()), parallels.WithLogger(logger))
 	case *api.Isolation_Container_:
+		containerPolicy := security.ContainerPolicy()
+		if containerPolicy == nil {
+			return nil, fmt.Errorf("%w: \"container\" isolation is not allowed by this Persistent Worker's "+
+				"security settings", ErrInvalidIsolation)
+		}
+
 		return container.New(iso.Container.Image, iso.Container.Cpu, iso.Container.Memory, iso.Container.Volumes)
 	case *api.Isolation_Tart_:
+		tartPolicy := security.TartPolicy()
+		if tartPolicy == nil {
+			return nil, fmt.Errorf("%w: \"tart\" isolation is not allowed by this Persistent Worker's "+
+				"security settings", ErrInvalidIsolation)
+		}
+
+		if !tartPolicy.ImageAllowed(iso.Tart.Image) {
+			return nil, fmt.Errorf("%w: Tart VM image %q is not allowed by this Persistent Worker's "+
```

Comment:
`Tart` is quoted as `\"tart\"` above. Maybe also add quotes around `%q`?

Reply:
You mean change `Tart` to `\"tart\"`? Because `%q` used for the image already [provides quoting](https://pkg.go.dev/fmt).

Reply:
> You mean change Tart to \"tart\"?

Yes.

> Because %q used for the image already [provides quoting](https://pkg.go.dev/fmt).

Sorry - I'm not a go dev at all...